internal/provider/data_source_test.go

```diff
@@ -83,124 +35,62 @@ func TestDataSource_http404(t *testing.T) {
 	defer testHttpMock.server.Close()
 
 	resource.UnitTest(t, resource.TestCase{
-		Providers: testProviders,
+		ProviderFactories: testProviders(),
 		Steps: []resource.TestStep{
 			{
-				Config:      fmt.Sprintf(testDataSourceConfig_basic, testHttpMock.server.URL, 404),
+				Config:      fmt.Sprintf(testDataSourceConfigBasic, testHttpMock.server.URL, 404),
 				ExpectError: regexp.MustCompile("HTTP request error. Response code: 404"),
 			},
 		},
 	})
 }
 
-const testDataSourceConfig_withHeaders = `
-data "http" "http_test" {
-  url = "%s/restricted/meta_%d.txt"
-
-  request_headers = {
-    "Authorization" = "Zm9vOmJhcg=="
-  }
-}
-
-output "body" {
-  value = data.http.http_test.body
-}
-`
-
 func TestDataSource_withHeaders200(t *testing.T) {
 	testHttpMock := setUpMockHttpServer()
 
 	defer testHttpMock.server.Close()
 
 	resource.UnitTest(t, resource.TestCase{
-		Providers: testProviders,
+		ProviderFactories: testProviders(),
 		Steps: []resource.TestStep{
 			{
-				Config: fmt.Sprintf(testDataSourceConfig_withHeaders, testHttpMock.server.URL, 200),
-				Check: func(s *terraform.State) error {
-					_, ok := s.RootModule().Resources["data.http.http_test"]
-					if !ok {
-						return fmt.Errorf("missing data resource")
-					}
-
-					outputs := s.RootModule().Outputs
-
-					if outputs["body"].Value != "1.0.0" {
-						return fmt.Errorf(
-							`'body' output is %s; want '1.0.0'`,
-							outputs["body"].Value,
-						)
-					}
-
-					return nil
-				},
+				Config: fmt.Sprintf(testDataSourceConfigWithHeaders, testHttpMock.server.URL, 200),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceHeaders("data.http.http_test"),
```

Comment:
Seems like this could be written as the below using the acceptance testing helper functions, rather than introducing indirection with a separate function:

```go
resource.TestCheckResourceAttr("data.http.http_test", "body", "1.0.0"),
```

Reply:
Fixed.